eclair-core/src/test/scala/fr/acinq/eclair/io/PeerSpec.scala

```diff
@@ -136,6 +136,20 @@ class PeerSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike with Paralle
     mockServer.close()
   }
 
+  test("return connection failure for a peer with an invalid dns host name") { f =>
+    import f._
+
+    // this actor listens to connection requests and creates connections
+    system.actorOf(ClientSpawner.props(nodeParams.keyPair, nodeParams.socksProxy_opt, nodeParams.peerConnectionConf, TestProbe().ref, TestProbe().ref))
+
+    val invalidDnsHostname_opt = NodeAddress.fromParts("eclair.invalid", 9735).toOption
```

Comment:
You can also add an assertion to verify that this is correctly parsed:

```scala
assert(invalidDnsHostname_opt.nonEmpty)
```

Reply:
fixed in [505c7cb](https://github.com/ACINQ/eclair/pull/2234/commits/505c7cb0442093fb134d503dde7689dfee5f8692)